src/lib/importContext.ts

```diff
@@ -0,0 +1,88 @@
+import { existsSync, readdir, readFile, readJson, stat } from "fs-extra";
+import path = require("path");
+
+export class ImportContext {
+	public packageJson: any;
+	public ioPackageJson: any;
+
+	constructor(private readonly baseDir: string) {
+		console.log(`Importing from ${baseDir}`);
+	}
+
+	public async load(): Promise<void> {
+		this.packageJson = await this.readJsonFile("package.json");
+		this.ioPackageJson = await this.readJsonFile("io-package.json");
+	}
+
+	public async readJsonFile<T>(fileName: string): Promise<T> {
+		return (await readJson(path.join(this.baseDir, fileName))) as T;
+	}
+
+	public async directoryExists(dirName: string): Promise<boolean> {
+		const fullPath = path.join(this.baseDir, dirName);
+		return existsSync(fullPath) && (await stat(fullPath)).isDirectory();
+	}
+
+	public async fileExists(dirName: string): Promise<boolean> {
+		const fullPath = path.join(this.baseDir, dirName);
+		return existsSync(fullPath) && (await stat(fullPath)).isFile();
+	}
+
+	public async hasFilesWithExtension(
+		dirName: string,
+		extension: string,
+	): Promise<boolean> {
+		return (
+			(await this.directoryExists(dirName)) &&
+			!!(await readdir(path.join(this.baseDir, dirName))).find((f) =>
+				f.toLowerCase().endsWith(extension.toLowerCase()),
+			)
+		);
+	}
+
+	public hasDevDependency(packageName: string): boolean {
+		return (
+			this.packageJson.devDependencies &&
+			this.packageJson.devDependencies.hasOwnProperty(packageName)
+		);
+	}
+
+	public async getMainFileContent(): Promise<string> {
+		if (
+			!this.packageJson.main ||
+			!this.packageJson.main.endsWith(".js") ||
+			!(await this.fileExists(this.packageJson.main))
+		) {
+			// we don't have a main JavaScript file, it will be impossible to find code
+			return "";
+		}
+
+		try {
+			const tsMain = path.join(
+				this.baseDir,
+				"src",
+				this.packageJson.main.replace(/\.js$/, ".ts"),
+			);
+			if (existsSync(tsMain)) {
+				// most probably TypeScript
+				return await readFile(tsMain, { encoding: "utf8" });
+			} else {
+				return await readFile(
```

Comment:
```suggestion
				return readFile(
```

Reply:
readFile is async, or why would you remove the await?

Reply:
see other comment